src/UI/Components/Diagram/components/FormModal.tsx

```diff
@@ -178,7 +174,7 @@ const FormModal = ({
       return values;
     };
 
-    const tempPossibleForms = getPossibleForms(services, [
+    const tempPossibleForms = getPossibleForms(serviceModels, [
```

Comment:
missing some comments in this file to add to the reasoning behind the code.

Reply:
it will be all reworked